treesit-fold-parsers.el

```diff
@@ -217,6 +217,8 @@
   "Rule set for Elisp."
   '((macro_definition    . treesit-fold-range-elisp-function)
     (function_definition . treesit-fold-range-elisp-function)
+    (special_form   . treesit-fold-range-seq)
+    (list           . treesit-fold-range-seq)
```

Comment:
`list` and `special_form` are almost everything, so I didn't include them in the first place. 🤔

Reply:
as far as i could test, it doesn't conflict with `defun`s and `defmacro`s, 
It can be useful for fold large lists, or large conditionals

Reply:
> as far as i could test, it doesn't conflict with defuns and defmacros,

Yes, you are right. :)

> It can be useful in some cases such as fold large lists, or large conditionals

My concern is most of the statements are list. Like `(require 'xxx)` is also identified as a list. 🤔

Reply:
> > It can be useful in some cases such as fold large lists, or large conditionals
> 
> My concern is most of the statements are list. Like `(require 'xxx)` is also identified as a list. 🤔

it only would get folded if it's something like:
```elisp
(require 'subr-x
         nil
         t)
```
which i think it would be very rare that someone indent it like that.

Reply:
I guess we can try and see if this work. I am a bit worry on the indicator's performance. 🤔

Reply:
I think we still need some improvements, for example:

```elisp
(define-minor-mode cogru-mode
  "Minor mode `cogru-mode'."
  :global t
  :require 'cogru-mode
  :group 'cogru
  (if cogru-mode (cogru-mode--enable) (cogru-mode--disable)))
```

will fold into:

```elisp
(...)
```

It will be great if it can fold like `defun`, etc.

```elisp
(define-minor-mode cogru-mode
  ...)
```

`defvar`, `defcustom`, `defvar-local` all have the same issue. 🤔

Reply:
> Feel free to split this into smaller PRs. It will be much easier for reviewers to review. :)

You are right, i will remove the elisp parser changes for a separated PR.

I apologize if i merged these commits in 1 PR, most of these changes would generate incompatibility each to other.